synapseutils/sync.py

```diff
@@ -256,22 +261,39 @@ def sync(self, entity, path, ifcollision, followLink, downloadFile=True, manifes
         files.sort(key=lambda f: f.get('path') or '')
         return files
 
-    def _sync_file(self, entity_id, parent_folder_sync, path, ifcollision, followLink, progress, downloadFile):
+    def _sync_file(self, entity_id, parent_folder_sync, path, ifcollision, followLink, progress, downloadFile, onlyMatching):
         try:
             # we use syn.get to download the File.
             # these context managers ensure that we are using some shared state
             # when conducting that download (shared progress bar, ExecutorService shared
             # by all multi threaded downloads in this sync)
             with progress.accumulate_progress(), \
                     download_shared_executor(self._executor):
-
-                entity = self._syn.get(
-                    entity_id,
-                    downloadLocation=path,
-                    ifcollision=ifcollision,
-                    followLink=followLink,
-                    downloadFile=downloadFile,
-                )
+            
+                file_matches = True
+                if onlyMatching is not None:
```

Comment:
**Nit (_i.e._ [optional](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.)):** This can be simplified to `if onlyMatching:` or perhaps more defensively as `if isinstance(onlyMatching, re.Pattern)` if we expect pre-compiled regex objects.